tests/Module/Fixture/Events.php

```diff
@@ -294,26 +286,88 @@ public function createEventsWithRelatedUserAndUser(int $owner, array $events)
             ];
 
             $eventId = $this->laravel->haveRecord(self::TABLE_EVENTS, $recordEvents);
-            $this->laravel->seeRecord(self::TABLE_EVENTS, $recordEvents);
 
             $recordRelatedUsers = [
-                'owner_id' => $owner,
+                'owner_id' => $ownerId,
                 'event_id' => $eventId,
                 'user_id' => $userId ?? '',
                 'related_user_id' => $temporaryUserId ?? '',
                 'created_at' => $createdAt,
             ];
-            
+
             $this->laravel->haveRecord(self::TABLE_RELATED_USERS, $recordRelatedUsers);
-            $this->laravel->seeRecord(self::TABLE_RELATED_USERS, $recordRelatedUsers);
 
             $users[] = $userId ?? $temporaryUserId;
             unset($userId);
             unset($temporaryUserId);
         }
     }
 
-    public function createEventsWithRelatedUserAndUserForExperimentStats(int $owner, array $events)
+    public function createRevenueEventsWithRelatedUserAndUser(
+        int $ownerId,
+        array $events,
+        int $eventsCount,
+        string $typeEvents,
+        ?string $staticUserId = null,
+        ?string $staticTemporaryUserId = null
+    ) {
+        $users = [];
+
+        $filterEvents = array_values(array_filter($events, function ($event) use ($typeEvents) {
+           return $event['type'] === $typeEvents;
+        }));
+
+        $n = 0;
+
+        foreach(range(1, $eventsCount) as $i) {
+            if ($n === count($filterEvents)) {
+                $n = 0;
```

Comment:
completely mad. You don't need to do this, it's unpredictable. When you have handled the full array, you're going to the first position again. 
It's not declarative and not predictable.